### ISSUE TITLE
`expect-expect` to consider nested `expect` in functions

### ISSUE THREAD

**Issue by AntonioVentilii** [vitest-dev/eslint-plugin-vitest#697]
Referring to issue https://github.com/vitest-dev/eslint-plugin-vitest/issues/696 , I would like to ask the feature to check into the functions called in the test? I imagine it is a common pattern to call functions that contain tests, so that there is no expect inside an it(...) statement.

```ts
export const testDerivedUpdates = async (...) => {
....
	derivedMocks.forEach((mockFn) => expect(mockFn).toHaveBeenCalledTimes(2));
....
};


it('should ......', async () => {
	await testDerivedUpdates(...);
});
```

**Comment by veritem** [comment]
You can always disable the expect-expect rule; I don't think this is a general use case.

**Comment by AntonioVentilii** [comment]
@veritem 

1. The idea is to use the rule not turn it off.
2. I actually see it as a common pattern to use functions with pre-created tests and just provide params. It is quite used around a few project. Agreed, they don't use this plugin, but maybe that may be a reason? This rule is straight in the recommended suite, so...
